### PR TITLE
[WFLY-18448] Update SmallRye OpenTelemetry to 2.5.1

### PR DESCRIPTION
**Pull request by jasondlee**
https://issues.redhat.com/browse/WFLY-18448

Bump SR Otel to 2.5.1
Update modules to reflect upstream changes

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -397,8 +397,9 @@
         <version.io.agroal>2.0</version.io.agroal>
         <version.io.micrometer>1.9.13</version.io.micrometer>
         <version.io.netty>4.1.100.Final</version.io.netty>
-        <version.io.opentelemetry.opentelemetry>1.20.0</version.io.opentelemetry.opentelemetry>
-        <version.io.opentelemetry.proto>0.19.0-alpha</version.io.opentelemetry.proto>
+        <version.io.opentelemetry.opentelemetry>1.29.0</version.io.opentelemetry.opentelemetry>
+        <version.io.opentelemetry.proto>0.20.0-alpha</version.io.opentelemetry.proto>
+        <version.io.opentelemetry.opentelemetry-semconv>1.21.0-alpha</version.io.opentelemetry.opentelemetry-semconv>
```
Comment (jamezp) [inline pom.xml:402]: Is there any indication these will come out of alpha? I know we've been shipping with alphas, but I'm just curious.

File: testsuite/integration/microprofile-tck/telemetry/src/test/java/io/smallrye/opentelemetry/arquillian/DeploymentProcessor.java [hunk 2]
```diff
@@ -30,7 +31,9 @@ public void process(Archive<?> archive, TestClass testClass) {
             extensionsJar.addAsServiceProvider(Providers.class, ExceptionMapper.class);
 
             WebArchive war = WebArchive.class.cast(archive);
-            war.addAsLibraries(extensionsJar);
+            war
+                    .addAsLibraries(extensionsJar)
+                    .addAsWebInfResource(new StringAsset("<beans bean-discovery-mode=\"all\"/>"), "beans.xml");
```
Comment (jamezp) [inline testsuite/integration/microprofile-tck/telemetry/src/test/java/io/smallrye/opentelemetry/arquillian/DeploymentProcessor.java:36]: Not saying a -1 to this, but what was the reason for this being required? It feels like a bit of a bug, but I could be wrong.

Comment (jasondlee) [reply testsuite/integration/microprofile-tck/telemetry/src/test/java/io/smallrye/opentelemetry/arquillian/DeploymentProcessor.java:36]: Without it, the injection is not performed in three tests for reasons I can't explain. I should, perhaps, see what changed in those tests.

File: pom.xml (hunk 1 above)
Comment (jasondlee) [reply pom.xml:402]: I have not seen any indication, no.

**Review by jamezp — APPROVED** [review]

**Comment by jamezp** [comment]
LGTM

**Pull request merged**